pkg/skaffold/verify/k8sjob/verify.go

```diff
@@ -0,0 +1,461 @@
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/kubectl/pkg/scheme"
+
+	component "github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/deploy/component/kubernetes"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/deploy/kubectl"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/deploy/label"
+	eventV2 "github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/event/v2"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/graph"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/instrumentation"
+	k8sjoblogger "github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/k8sjob/logger"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/k8sjob/tracker"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/kubernetes"
+	kubernetesclient "github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/kubernetes/client"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/loader"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/log"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/schema/latest"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/status"
+	"github.com/GoogleContainerTools/skaffold/v2/pkg/skaffold/util"
+)
+
+// Verifier verifies deployments using kubernetes libs/CLI.
+type Verifier struct {
+	configName string
+
+	cfg              []*latest.KubernetesClusterVerifyTestCase
+	tracker          *tracker.JobTracker
+	imageLoader      loader.ImageLoader
+	logger           *k8sjoblogger.Logger
+	statusMonitor    status.Monitor
+	originalImages   []graph.Artifact // the set of images parsed from the Verifier's manifest set
+	localImages      []graph.Artifact // the set of images marked as "local" by the Runner
+	kubectl          kubectl.CLI
+	labeller         *label.DefaultLabeller
+	defaultNamespace string
+}
+
+// NewVerifier returns a new Verifier for a VerifyConfig filled
+// with the needed configuration for `kubectl apply`
+func NewVerifier(ctx context.Context, cfg kubectl.Config, labeller *label.DefaultLabeller, v *latest.KubernetesClusterVerifier, artifacts []*latest.Artifact) (*Verifier, error) {
+	defaultNamespace := ""
+	b, err := (&util.Commander{}).RunCmdOut(context.Background(), exec.Command("kubectl", "config", "view", "--minify", "-o", "jsonpath='{..namespace}'"))
+	if err == nil {
+		defaultNamespace = strings.Trim(string(b), "'")
+		if defaultNamespace == "default" {
+			defaultNamespace = ""
+		}
+	}
+	if v.DefaultNamespace != nil {
+		var err error
+		defaultNamespace, err = util.ExpandEnvTemplate(*v.DefaultNamespace, nil)
+		if err != nil {
+			return nil, err
+		}
+	}
+	kubectl := kubectl.NewCLI(cfg, v.Flags, defaultNamespace)
+	// default namespace must be "default" not "" when used to create and stream logs from Job(s)
+	if defaultNamespace == "" {
+		defaultNamespace = "default"
+	}
+	tracker := tracker.NewContainerTracker()
+	logger, err := k8sjoblogger.NewLogger(ctx, tracker, labeller, kubectl.KubeContext)
```

Comment:
Right now, the `k8sjoblogger.NewLogger`, returns always nil as error, is it possible to remove the returned error?

Reply:
done